### PR TITLE
Fixes for django xblock client

### PR DESCRIPTION
**Pull request by cpennington**

### CODE REVIEW

**Comment by cpennington** [comment]
@alawibaba @utkjad @macdiesel @doctoryes: Review?

**Comment by doctoryes** [comment]
:+1:

File: edx_user_state_client/tests.py [hunk 1]
```diff
@@ -33,19 +38,37 @@ class _UserStateClientTestUtils(TestCase):
...
 
-    def get(self, user_idx, block_idx, fields=None):
+    @contract(block=int)
+    def _block_type(self, block):  # pylint: disable=unused-argument
+        """Return the block type for the specified ``block_idx``."""
+        return 'block_type'
+
+    @contract(course=int)
+    def _course(self, course):
+        """Return a CourseKey for the course ``course_idx``"""
```
Comment (macdiesel) [inline edx_user_state_client/tests.py]: course_idx should be course.

File: edx_user_state_client/tests.py [hunk 2]
```diff
@@ -33,19 +38,37 @@ class _UserStateClientTestUtils(TestCase):
...
-            'block{}'.format(block_idx)
+            self._course(course),
+            self._block_type(block),
+            'block{}'.format(block)
         )
 
-    def get(self, user_idx, block_idx, fields=None):
+    @contract(block=int)
+    def _block_type(self, block):  # pylint: disable=unused-argument
+        """Return the block type for the specified ``block_idx``."""
```
Comment (macdiesel) [inline edx_user_state_client/tests.py]: nit: should be block not block_idx

**Comment by macdiesel** [comment]
Other than those 2 doc nitpicks I'm a :+1: solid stuff. I like the new semantics.

**Comment by alawibaba** [comment]
:+1:

**Pull request merged**